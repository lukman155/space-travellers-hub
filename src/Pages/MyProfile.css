.profile {
  display: flex;
  justify-content: flex-start;
  padding: 20px;
  margin: 0 auto;
}

.myContainer {
  padding: 20px;
  width: 80%;
}

.heading {
  font-size: 20px;
  font-weight: 600;
  color: #000;
  margin-bottom: 20px;
}

.profileList {
  list-style: none;
}

.profileList li {
  padding: 20px;
  border: 1px solid #ccc;
  display: flex;
  justify-content: space-between;
}

.leaveMissionButton {
  height: 30px;
  text-align: center;
  font-size: 15px;
  padding: 1%;
  width: 40%;
  background-color: white;
  color: rgb(88, 87, 87);
  border: 2px solid rgb(88, 87, 87);
  border-radius: 5px;
  align-self: end;
}

.profileList li > a {
  color: rgb(88, 87, 87);
  text-decoration: none;
}

.profileList li > a:hover {
  color: rgb(38, 85, 155);
}

.profileList li > div {
  flex-grow: 2;
}

.leaveMissionButton:hover {
  color: rgb(238, 26, 26);
  border: 2px solid rgb(238, 26, 26);
  cursor: pointer;
}
