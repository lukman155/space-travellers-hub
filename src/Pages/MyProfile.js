import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { joinMission } from '../Redux/Mission/mission';
import { toggleReserve } from '../Redux/rockets/rocketsSlice';
import './MyProfile.css';

const MyProfile = () => {
  const dispatch = useDispatch();
  const joinedMissions = useSelector((state) => state.missions).filter((missions) => missions.join);
  const joinedMission = joinedMissions.map((mission) => (
    <li key={mission.mission_id}>
      <a
        href={mission.wikipedia}
        target="_blank"
        rel="noreferrer"
      >
        {mission.mission_name}
      </a>

      <button
        type="button"
        className="leaveMissionButton"
        onClick={() => dispatch(joinMission(mission.mission_id))}
      >
        Leave Mission
      </button>

    </li>
  ));

  const reservedRockets = useSelector((state) => (
    state.rockets).filter((rockets) => rockets.reserved));
  const reservedRocket = reservedRockets.map((rocket) => (
    <li key={rocket.rocket_id}>
      <a
        href={rocket.wikipedia}
        target="_blank"
        rel="noreferrer"
      >
        {rocket.rocket_name}
      </a>

      <button
        type="button"
        className="leaveMissionButton"
        onClick={() => dispatch(toggleReserve(rocket.id))}
      >
        Cancel Reservation
      </button>

    </li>

  ));
  return (
    <div className="profile">
      <div className="myContainer">
        <div className="heading">My Missions</div>
        <ul className="profileList">
          {joinedMissions.length > 0 ? (joinedMission)
            : (<p>No Missions Joined</p>)}
        </ul>
      </div>
      <div className="myContainer">
        <div className="heading">My Rockets</div>
        <ul className="profileList">
          {reservedRockets.length > 0 ? (reservedRocket)
            : (<p>No Rockets Reserved</p>)}
        </ul>
      </div>
    </div>
  );
};

export default MyProfile;
