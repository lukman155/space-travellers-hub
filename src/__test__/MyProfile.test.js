import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import MyProfile from '../Pages/MyProfile';

describe('MyProfile', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render MyProfile component', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.getByText('My Missions')).toBeInTheDocument();
    expect(screen.getByText('My Rockets')).toBeInTheDocument();
  });
});
