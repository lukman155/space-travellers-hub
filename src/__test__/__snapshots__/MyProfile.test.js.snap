// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile renders correctly 1`] = `
<div
  className="profile"
>
  <div
    className="myContainer"
  >
    <div
      className="heading"
    >
      My Missions
    </div>
    <ul
      className="profileList"
    >
      <p>
        No Missions Joined
      </p>
    </ul>
  </div>
  <div
    className="myContainer"
  >
    <div
      className="heading"
    >
      My Rockets
    </div>
    <ul
      className="profileList"
    >
      <p>
        No Rockets Reserved
      </p>
    </ul>
  </div>
</div>
`;
